File: app/src/main/java/lt/markmerkk/utils/tracker/GATracker.java
package lt.markmerkk.utils.tracker;

import com.brsanthu.googleanalytics.AppViewHit;
import com.brsanthu.googleanalytics.EventHit;
import com.brsanthu.googleanalytics.GoogleAnalytics;
import com.brsanthu.googleanalytics.PageViewHit;
import com.google.common.base.Strings;
import lt.markmerkk.Main;
import lt.markmerkk.utils.tracker.interfaces.ITracker;

/**
 * Created by mariusmerkevicius on 3/11/16.
 * Represents tracker for tracking various events through {@link GoogleAnalytics}
 */
public class GATracker implements ITracker {
  final GoogleAnalytics analytics;

  public GATracker() {
    if (Main.GA_KEY == null)
      throw new IllegalArgumentException("ga_key == null");
    analytics = new GoogleAnalytics(Main.GA_KEY, Main.APP_NAME, Main.VERSION_NAME);
  }

  @Override
  public void sendEvent(String category, String action, String label, int value) {
    if (Strings.isNullOrEmpty(category)) return;
    if (Strings.isNullOrEmpty(action)) return;
    if (Strings.isNullOrEmpty(label)) return;
    analytics.postAsync(new EventHit(category, action, label, value));
  }

  @Override
  public void sendEvent(String category, String action) {
    if (Strings.isNullOrEmpty(category)) return;
    if (Strings.isNullOrEmpty(action)) return;
    analytics.postAsync(new EventHit(category, action));
  }

  public void sendView(String contentDescription) {
    if (Strings.isNullOrEmpty(contentDescription)) return;
    analytics.postAsync(new AppViewHit(Main.APP_NAME, Main.VERSION_NAME, contentDescription));
  }

  @Override
  public void stop() {
    analytics.close();
  }
}

File: app/src/main/java/lt/markmerkk/utils/tracker/GATracker.kt
package lt.markmerkk.utils.tracker

import com.brsanthu.googleanalytics.AppViewHit
import com.brsanthu.googleanalytics.EventHit
import com.brsanthu.googleanalytics.GoogleAnalytics
import lt.markmerkk.Config
import lt.markmerkk.Main

/**
 * Created by mariusmerkevicius on 3/11/16.
 * Represents tracker for tracking various events through [GoogleAnalytics]
 */
class GATracker(
        private val config: Config
) : ITracker {
    private val analytics: GoogleAnalytics

    init {
        analytics = GoogleAnalytics(config.gaKey, Main.APP_NAME, config.versionName)
    }

    override fun sendEvent(category: String, action: String, label: String, value: Int) {
        if (category.isNullOrEmpty()) return
        if (action.isNullOrEmpty()) return
        if (label.isNullOrEmpty()) return
        analytics.postAsync(EventHit(category, action, label, value))
    }

    override fun sendEvent(category: String, action: String) {
        if (category.isNullOrEmpty()) return
        if (action.isNullOrEmpty()) return
        analytics.postAsync(EventHit(category, action))
    }

    override fun sendView(contentDescription: String) {
        if (contentDescription.isNullOrEmpty()) return
        analytics.postAsync(AppViewHit(Main.APP_NAME, config.versionName, contentDescription))
    }

    override fun stop() {
        analytics.close()
    }
}
